plumbing/protocol/packp/capability/list.go

```diff
@@ -0,0 +1,158 @@
+package capability
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+var (
+	// ErrUnknownCapability is returned if a unknown capability is given
+	ErrUnknownCapability = errors.New("unknown capability")
+	// ErrArgumentsRequired is returned if no arguments are giving with a
+	// capability that requires arguments
+	ErrArgumentsRequired = errors.New("arguments required")
+	// ErrArguments is returned if arguments are given with a capabilities that
+	// not supports arguments
+	ErrArguments = errors.New("arguments not allowed")
+	// ErrEmtpyArgument is returned when an empty value is given
+	ErrEmtpyArgument = errors.New("empty argument")
+	// ErrMultipleArguments multiple argument given to a capabilities that not
+	// support it
+	ErrMultipleArguments = errors.New("multiple arguments not allowed")
+)
+
+// List represents a list of capabilities
+type List struct {
+	m map[Capability]*entry
+	o []string
+}
+
+type entry struct {
```

Comment:
This must be public, as it is what the clients would like to use to add new capabilities, instead of a `Capability` plus some values.

I would rename `entry` to `Capability` and rename `Capability` to `Type`.